ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -492,6 +498,59 @@ const DiskSizeSelector = ({onChange, disabled, selectedDiskSize, diskSize, idPre
     </FlexRow>;
 };
 
+const GpuConfigSelector = ({disabled, onChange, selectedMachine, gpuConfig})  => {
+  const {
+    gpuType = 'nvidia-tesla-t4',
+    numOfGpus = 1,
+  } = gpuConfig || {};
+  const [selectedGpuType, setSelectedGpuType] = useState<string>(gpuType);
+  const [selectedNumOfGpus, setSelectedNumOfGpus] = useState<number>(numOfGpus);
+  const validGpuOptions = getValidGpuTypes(selectedMachine.cpu, selectedMachine.memory);
+  const validGpuNames = fp.flow(fp.map('name'), fp.uniq, fp.sortBy('price'))(validGpuOptions);
+  const validNumGpusOptions = fp.flow(fp.filter({ type: selectedGpuType }), fp.map('numGpus'))(validGpuOptions);
+  const [enableGpu, setEnableGpu] = useState<boolean>(!!gpuConfig);
```

Comment:
nit: I might move this to the top of the function, or next to the other hooks

Reply:
Hi, CH. Do you mean moving these variables out of the function? I put them inside the function since it's just used here as local variables of this function. It looks cleaner. And I'm not sure wether the compiler will actually move this code block outside?

Reply:
I'm suggesting moving it up to L521 within this function. `useState` is a "React hook", these have special requirements and generally it's preferable to define these before anything else (we have exceptions to this); defining it next to the other hooks in this component would also make sense.